packages/hash/frontend/src/components/layout/PageSidebar/AccountEntityTypeList/AccountEntityTypeList.tsx

```diff
@@ -18,132 +104,141 @@ export const AccountEntityTypeList: VFC<AccountEntityTypeListProps> = ({
+  const [searchQuery, setSearchQuery] = useState("");
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const sortActionsPopupState = usePopupState({
+    variant: "popover",
+    popupId: "type-sort-actions-menu",
+  });
+
+  useEffect(() => {
+    if (searchVisible) {
+      searchInputRef.current?.focus();
+    }
+  }, [searchVisible]);
+
+  // todo: handle search server side
+  const filteredData = useMemo(() => {
+    let entityTypes = data?.getAccountEntityTypes ?? [];
+
+    if (searchQuery) {
+      entityTypes = entityTypes.filter(({ properties }) =>
+        properties.title.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
+    }
+
+    // Right now we just handle ascending/descending
+    if (sortType === "asc" || sortType === "desc") {
+      return orderBy(entityTypes, ["properties.title"], [sortType]);
+    }
+
+    return entityTypes;
+  }, [sortType, data, searchQuery]);
 
   return (
     <Box>
       <NavLink
         title="Types"
-        endAdornment={<CreateEntityTypeButton accountId={accountId} />}
+        endAdornmentProps={{
+          tooltipTitle: "Create new type",
+          onClick: () => router.push(`/${accountId}/types/new`),
+          "data-testid": "create-entity-btn",
+        }}
       >
         <Box component="ul">
-          {/* Can be uncommented once we have a page that displays all entity types */}
-          {/* <Box
+          <Box
+            tabIndex={0}
             component="li"
-            sx={{
+            sx={({ palette }) => ({
               display: "flex",
               alignItems: "center",
               mx: 0.5,
-              pl: 3.75,
-            }}
+              minHeight: 36,
+              borderRadius: "4px",
+              "&:hover": {
+                backgroundColor: palette.gray[20],
+              },
+              "&:focus-visible": {
+                backgroundColor: "red",
+              },
+              ...(sortActionsPopupState.isOpen && {
+                backgroundColor: palette.gray[20],
+              }),
+            })}
           >
-            <Typography
-              variant="smallTextLabels"
-              fontWeight="600"
-              sx={{ mr: "auto", color: ({ palette }) => palette.gray[80] }}
+            <Box
+              display="flex"
+              alignItems="center"
+              flex={1}
+              mr={1.25}
+              pl={3.75}
+              position="relative"
             >
-              View All Types
-            </Typography>
-            <IconButton sx={{ mr: 1.25 }}>
-              <FontAwesomeIcon icon={faSearch} />
-            </IconButton>
-            <IconButton onClick={toggleSort}>
-              <FontAwesomeIcon icon={faArrowUpAZ} />
-            </IconButton>
-          </Box> */}
-          {sortedData.map((entityType) => {
-            return (
-              <Box
-                component="li"
+              <Link
+                href="/"
```

Comment:
We currently don't have a page for viewing all types. Once we do, we can update this